qutip/solver/flimesolve.py

```
@@ -0,0 +1,646 @@
+    '''
+    Hdim = len(floquet_basis.e_quasi)  # Dimensionality of the Hamiltonian
+    args_key = list(args.values())[0]  # Frequency dependence of Hamiltonian
+
+    # Forming tlist to take FFT of collapse operators
+    time = T
+    dt = time / Nt
+    tlist = np.linspace(0, time - dt, Nt)
+
+    total_R_tensor = {}
+    for cdx, c_op in enumerate(c_ops):
+        '''
+        Transforming the lowering operator into the Floquet Basis
+            and taking the FFT
+        '''
+        c_op_Floquet_basis = np.array(
+            [floquet_basis.to_floquet_basis(c_op, t).full() for t in tlist])
+
+        c_op_Fourier_amplitudes_list = np.array(
+            np.fft.fft(c_op_Floquet_basis, axis=0) / len(tlist)
+        )
+
+        '''
+        Next step is to form arrays of the rate products (rate_products)
+            and the frequency at which these components rotate (delta_array),
+            then dividing the arrays to form the rate quotient array.
+        
+        In quotient array, setting any undefined value, i.e. those for which
+            rate_products = 0, equal to time_sense+1 so that the mask
+            arrays mask out these values.
+        '''
+        rate_products = np.einsum('lab,kcd->abcdlk',
+                                  c_op_Fourier_amplitudes_list,
+                                  np.conj(c_op_Fourier_amplitudes_list))
+
+        delta_array = np.add.outer(
+            np.subtract.outer(
+                np.subtract.outer(floquet_basis.e_quasi,
+                                  floquet_basis.e_quasi),
+                np.subtract.outer(floquet_basis.e_quasi, floquet_basis.e_quasi)
+            )/args_key,
+            np.subtract.outer(np.arange(Nt), np.arange(Nt))
+        )
+
+        quotient_array = np.divide(
+            args_key*delta_array,
+            abs(rate_products),
+            out=np.full(np.shape(delta_array), (time_sense+1), dtype=complex),
```

Let's perhaps add a comment here explaining that the `out` value is set to provide defaults outside of the quotient mask range where the rate_products are zero.